api/src/infrastructure/docker.rs

```diff
@@ -85,6 +85,53 @@ impl DockerInfrastructure {
         DockerInfrastructure {}
     }
 
+    async fn find_deployment_task_container(
+        &self,
+        deployment_id: &String,
+    ) -> Result<Option<Container>, ShipLiftError> {
+        let docker = Docker::new();
+        let containers = docker.containers();
+
+        let list_options = ContainerListOptions::builder().all().build();
+
+        Ok(containers
+            .list(&list_options)
+            .await?
+            .into_iter()
+            .filter(|c| match c.labels.get(DEPLOYMENT_ID) {
+                None => false,
+                Some(deployment) => deployment == deployment_id,
+            })
+            .next())
+    }
+
+    async fn create_deployment_task_container(
+        &self,
+        deployment_id: &String,
+        app_name: &str,
+    ) -> Result<ContainerDetails, Error> {
+        // TODO: what to if there is already an deployment
```

Comment:
We could resolve this TODO in conjunction with removing the deployment guard. Instead of using the deployment guard in `Apps` we could ask the infrastructure if there is a task running and than reject with 409 as it is the case right now. 

Then it should be safe to return an error here if there is a deployment running.

Reply:
I'm a bit afraid to remove the guard in apps, because we don't have the task-thing in kubernetes yet.

It should still be possible to check for a running task here, right?

Reply:
I added a check that … just errors if there's already a matching container 😅

Reply:
That should be part of the issue that I'll create (see [here](https://github.com/aixigo/PREvant/pull/55#discussion_r602309415)).